snow/consensus/snowman/topological.go

```diff
@@ -601,72 +585,81 @@ func (ts *Topological) vote(ctx context.Context, voteStack []votes) (ids.ID, err
 //
 // We accept a block once its parent's snowball instance has finalized
 // with it as the preference.
-func (ts *Topological) acceptPreferredChild(ctx context.Context, n *snowmanBlock) error {
-	// We are finalizing the block's child, so we need to get the preference
-	pref := n.sb.Preference()
+func (ts *Topological) tryAcceptPreferredChild(ctx context.Context, blk *snowmanBlock, siblings set.Set[ids.ID]) (bool, error) {
+	blkID := blk.blk.ID()
+	height := blk.blk.Height()
+	if err := blk.Verify(ctx); err != nil {
```

Comment:
I wonder if it makes sense to rate limit (somehow) how many times we try to verify the same block per second? It seems we could attempt this after each poll.